src/flint/gfp_fmpz_elem.jl

```diff
@@ -219,6 +219,23 @@ function ^(x::gfp_fmpz_elem, y::Int)
    return gfp_fmpz_elem(d, R)
 end
 
+function ^(x::gfp_fmpz_elem, y::fmpz)
+   R = parent(x)
+   if y < 0
+      x = inv(x)
+      y = -y
+   end
+   d = fmpz()
+   ccall((:fmpz_mod_pow_fmpz, libflint), Nothing,
+	 (Ref{fmpz}, Ref{fmpz}, Ref{fmpz}, Ref{fmpz_mod_ctx_struct}),
+						 d, x.data, y, R.ninv)
+   return gfp_fmpz_elem(d, R)
```

Comment:
```suggestion
function ^(x::gfp_fmpz_elem, y::fmpz)
   R = parent(x)
   z = R()
   if 0 == ccall((:fmpz_mod_pow_fmpz, libflint), Cint,
                 (Ref{fmpz}, Ref{fmpz}, Ref{fmpz}, Ref{fmpz_mod_ctx_struct}),
                 z.data, x.data, y, R.ninv)
      if iszero(x)
         throw(DivideError())
      else
         error("Impossible inverse")
      end
   end
   return z
end
```

Reply:
I am working on an ImpossibleInverse exception for AA, but for now, will this do? Are we supposed to throw a DivideError (and not an "impossible inverse") on division by zero?